arch/xtensa/include/esp32/memory_layout.h

```diff
@@ -72,9 +72,23 @@
  *
  * When an internal heap is enabled this region starts at an offset equal to
  * the size of the internal heap.
+ *
+ * The QEMU bootloader image is slightly different than the chip's one.
+ * The ROM on PRO and APP uses different regions for static data. In QEMU,
+ * however, we load only one ROM binary, taken from the PRO CPU, and it is
+ * used by both CPUs.  So, in QEMU, if we allocate this part early, it will
+ * be clobbered once the APP CPU starts.
+ * We can delay the allocation to when everything has started through the
+ * board_late_initiliaze hook, as is done for the APP data, however this
+ * should be fixed from QEMU side.  The following macros, then, just skip
+ * PRO CPU's data when a QEMU image generation is enabled with SMP.
  */
 
-#define HEAP_REGION2_START  0x3ffe0450
+#if defined(CONFIG_ESP32_QEMU_IMAGE) && defined(CONFIG_SMP)
```

Comment:
can you consider to introduce a separate kconfig for this?

maybe we can follow this precedent.
```
config TIVA_WITH_QEMU
        bool "Workaround for tiva with qemu"
        default n
```
how do you think?

Reply:
Isn't `CONFIG_ESP32_QEMU_IMAGE` enough?  Or you want another one that's going to be auto selected if `CONFIG_ESP32_QEMU_IMAGE` and `CONFIG_SMP` are enabled?

`CONFIG_ESP32_QEMU_IMAGE` is supposed to generate a QEMU image, so maybe here it's being used for more than that.

Reply:
i feel it's awkward to change the behavior with the output binary/image options. but it might be just me.

Reply:
I agree, it does feel awkward to me too.
The thing is this workaround has to be applied whenever a QEMU image is generated with SMP, otherwise the board won't boot.  So if we add a new one it will end up being auto selected.
I'm also operating with the assumption that this should be fixed in QEMU (don't know exactly when, though.)
What do you think?

Reply:
for me, a separate config is more convenient.

*  CONFIG_ESP32_WITH_QEMU without CONFIG_ESP32_QEMU_IMAGE makes sense when building qemu images manually from nuttx or nuttx.bin. (i almost always do this when using qemu for xtensa)
* CONFIG_ESP32_QEMU_IMAGE without CONFIG_ESP32_WITH_QEMU might makes sense when using a fixed qemu in future.

having said that, i can live with the single CONFIG_ESP32_QEMU_IMAGE config. it isn't a blocker of the PR at all.